sys/shell/cmds/nanocoap_vfs.c

```diff
@@ -169,19 +169,34 @@ static int _nanocoap_put_handler(int argc, char **argv)
     file = argv[1];
     url = argv[2];
 
+    /* append filename to remote dir */
     if (_is_dir(url)) {
         const char *basename = strrchr(file, '/');
         if (basename == NULL) {
-            return -EINVAL;
+            basename = file;
+        } else {
+            ++basename;
         }
+
         if (snprintf(buffer, sizeof(buffer), "%s%s",
-                     url, basename + 1) >= (int)sizeof(buffer)) {
+                     url, basename) >= (int)sizeof(buffer)) {
             puts("Constructed URI too long");
             return -ENOBUFS;
         }
         url = buffer;
     }
 
+    /* relative file path */
+    if (*file != '/') {
```

Comment:
I think that would make it more clear that inspecting the first character only is deliberate. (Again, nit) 
```suggestion
    if (file[0] != '/') {
```

Reply:
Ah, this if is also take if `file == '-'`. I think that is okay and can be ignored?

Reply:
This was added as a convenience. We already have `CONFIG_NCGET_DEFAULT_DATA_DIR` to store downloads from `ncget` when the destination is not specified - so you can do

    ncget coap://example.com/files/data.txt

and it will end up in `CONFIG_NCGET_DEFAULT_DATA_DIR`.
This is just the reverse: It allows you do to

    ncput foo.txt coap://example.com/in/

without typing out the whole path.